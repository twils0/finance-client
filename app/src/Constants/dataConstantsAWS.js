// eslint-disable-next-line
export const actionTypes = {
  SET_AWS_STATUS: 'DATA_AUTH_AWS_STATUS',
  SET_AWS_USER: 'DATA_AUTH_SET_AWS_USER',
  RESET_AWS_STATE: 'DATA_AUTH_RESET_AWS_STATE',
};
