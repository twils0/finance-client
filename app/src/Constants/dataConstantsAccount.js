export const actionTypes = {
  SET_ACCOUNT_STATUS: 'DATA_ACCOUNT_SET_ACCOUNT_STATUS',
  SET_FIELD: 'DATA_ACCOUNT_SET_FIELD',
  SET_FIELDS: 'DATA_ACCOUNT_SET_FIELDS',
  RESET_ACCOUNT_STATE: 'DATA_ACCOUNT_RESET_ACCOUNT_STATE',
};

export const statusNames = {
  AWS_FIELDS: 'accountStatusAWSFields',
  STRIPE_FIELDS: 'accountStatusStripeFields',
  UPDATE_STRIPE_FIELDS_TOKEN: 'accountStatusUpdateStripeFieldsToken',
  UPDATE_STRIPE_FIELDS_REQUEST: 'accountStatusUpdateStripeFieldsRequest',
  UPDATE_DB_FIELDS: 'accountStatusUpdateDBFields',
  UPDATE_AWS_FIELDS: 'accountStatusUpdateAWSFields',
};

export const fieldNames = {
  NAME: 'accountName',
  EMAIL: 'accountEmail',
  EMAIL_ADDITIONAL: 'accountEmailAdditional',
  PHONE: 'accountPhone',
  NAME_ON_CARD: 'accountNameOnCard',
  PROMO_CODE: 'accountPromoCode',
  PROMO_CODE_VALID: 'accountPromoCodeValid',
  CUSTOMER: 'accountCustomer',
};
