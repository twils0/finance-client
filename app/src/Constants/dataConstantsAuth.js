export const actionTypes = {
  SET_AUTHENTICATED: 'DATA_AUTH_SET_AUTHENTICATED',
  SET_REDIRECT_URL: 'DATA_AUTH_SET_REDIRECT_URL',
  SET_AUTH_STATUS: 'DATA_AUTH_SET_AUTH_STATUS',
  SET_CODE_TYPE: 'DATA_AUTH_SET_CODE_TYPE',
  RESET_AUTH_STATE: 'DATA_AUTH_RESET_AUTH_STATE',
};

export const statusNames = {
  LOGIN: 'authStatusLogin',
  SIGN_OUT_DEVICES: 'authStatusSignOutDevices',
  LOGIN_MFA: 'authStatusLoginMFA',
  SIGN_UP: 'authStatusSignUp',
  VERIFY_PHONE: 'authStatusVerifyPhone',
  VERIFY_PHONE_CODE: 'authStatusVerifyPhoneCode',
  VERIFY_EMAIL: 'authStatusVerifyEmail',
  VERIFY_EMAIL_LINK: 'authStatusVerifyEmailLink',
  FORGOT_PASSWORD: 'authStatusForgotPassword',
  RESET_PASSWORD: 'authStatusResetPassword',
  CHANGE_PASSWORD: 'authStatusChangePassword',
  DELETE_ACCOUNT: 'authStatusDeleteAccount',
  LOGOUT: 'authStatusLogout',
};

export const codeTypeNames = {
  VERIFY_PHONE: 'verifyPhone',
  VERIFY_EMAIL: 'verifyEmail',
  VERIFY_EMAIL_ADDITIONAL: 'verifyEmailAdditional',
};
