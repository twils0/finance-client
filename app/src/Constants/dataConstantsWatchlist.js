export const actionTypes = {
  SET_WATCHLIST_STATUS: 'DATA_WATCHLIST_SECURITIES_STATUS',
  SET_SECURITIES_CURRENT: 'DATA_WATCHLIST_SET_SECURITIES_CURRENT',
  SET_SECURITIES_LIST: 'DATA_WATCHLIST_SET_SECURITIES_LIST',
  SET_SECURITIES_ALL: 'DATA_WATCHLIST_SET_SECURITIES_ALL',
  SET_SECURITY: 'DATA_WATCHLIST_SET_SECURITY',
  SET_SECURITY_DATA: 'DATA_WATCHLIST_SET_SECURITY_DATA',
  SET_SECURITY_SUBSCRIBE: 'DATA_WATCHLIST_SET_SECURITY_SUBSCRIBE',
  RESET_WATCHLIST_STATE: 'DATA_ACCOUNT_RESET_WATCHLIST_STATE',
};

export const codeTypeNames = {
  CONFIRM_PHONE: 'confirmPhone',
  CONFIRM_EMAIL: 'confirmEmail',
};

export const statusNames = {
  SECURITIES: 'watchlistStatusSecurities',
  GET_SECURITY_DATA: 'watchlistStatusGetSecurityData',
  UPDATE_SECURITIES: 'watchlistStatusUpdateSecurities',
};
