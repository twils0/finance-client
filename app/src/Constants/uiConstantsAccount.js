export const actionTypes = {
  SET_CURRENT_FORM: 'ACCOUNT_SET_CURRENT_FORM',
  SET_FORM_HEIGHT: 'ACCOUNT_SET_FORM_HEIGHT',
  SET_FORM_EDIT: 'ACCOUNT_SET_FORM_EDIT',
  SET_INPUT_VALUE_ERROR: 'ACCOUNT_SET_INPUT_VALUE_ERROR',
  SET_BUTTON_TEXT: 'ACCOUNT_SET_BUTTON_TEXT',
  SET_BUTTON_VISIBLE: 'ACCOUNT_SET_BUTTON_VISIBLE',
};

export const formNames = {
  CODE: 'codeForm',
  PROFILE: 'profileForm',
  BILLING: 'billingForm',
  CHANGE_PASSWORD: 'changePasswordForm',
  DELETE_ACCOUNT: 'deleteAccountForm',
};

export const buttonNames = {
  PROFILE: 'profileButton',
  BILLING: 'billingButton',
  CHANGE_PASSWORD: 'changePasswordButton',
  DELETE_ACCOUNT: 'deleteAccountButton',
  FIRST: 'firstButton',
  SECOND: 'secondButton',
};

export const buttonTexts = {
  RESEND: 'Resend',
  SUBMIT: 'Submit',
  EDIT: 'Edit',
  UPDATE: 'Update',
  CANCEL: 'Cancel',
  DELETE: 'Delete',
  CONFIRM: 'Confirm',
};

export const inputNames = {
  [formNames.CODE]: {
    CODE_PHONE: 'codeCodePhone',
  },
  [formNames.PROFILE]: {
    NAME: 'profileName',
    EMAIL: 'profileEmail',
    EMAIL_ADDITIONAL: 'profileEmailAdditional',
    PHONE: 'profilePhone',
  },
  [formNames.BILLING]: {
    NAME_ON_CARD: 'billingNameOnCard',
    PROMO_CODE: 'billingPromoCode',
    STRIPE: 'billingStripe',
  },
  [formNames.CHANGE_PASSWORD]: {
    OLD_PASSWORD: 'changePasswordOldPassword',
    NEW_PASSWORD: 'changePasswordNewPassword',
    NEW_PASSWORD2: 'changePasswordNewPassword2',
  },
  [formNames.DELETE_ACCOUNT]: {
    CONFIRM: 'deleteAccountConfirm',
  },
};
