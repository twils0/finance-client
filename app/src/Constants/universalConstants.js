import stripeConfig from '../../../stripe.config.json';
import dbConfig from '../../../db.config.json';
import iexConfig from '../../../iex.config.json';

export const baseURLs = {
  STRIPE: stripeConfig.url,
  DB: dbConfig.url,
  IEX: iexConfig.urls.api,
  IEX_DEV: iexConfig.urls.dev,
  IEX_TERMS: iexConfig.urls.terms,
};

export const URLs = {
  CUSTOMERS: '/customers',
  USERS: '/users',
  EMAILS: '/emails',
  SECURITIES: '/securities',
  SECURITIES_SEARCH: '/securities/search',
  IEX_COMPANY: tickerCusip => `/stock/${tickerCusip}/company`,
  IEX_QUOTE: tickerCusip => `/stock/${tickerCusip}/quote`,
};

export const axiosConfig = {
  STRIPE: {
    baseURL: baseURLs.STRIPE,
    timeout: 30000,
  },
  DB: {
    baseURL: baseURLs.DB,
    timeout: 30000,
  },
  IEX: {
    baseURL: baseURLs.IEX,
    timeout: 30000,
  },
};

export const pathNames = {
  DEFAULT: '*',
  LOGIN: '/login',
  SIGN_UP: '/signup',
  CODE: '/code',
  CODE_VERIFY_EMAIL: '/code/:verificationId?',
  TERMS: '/terms',
  INTERNAL: '/internal',
  WATCHLIST: '/internal/watchlist',
  WATCHLIST_SECURITY_ID: '/internal/watchlist/:securityId?',
  ACCOUNT: '/internal/account',
};

export const requestStatusTypes = {
  IDLE: 'IDLE',
  LOADING: 'LOADING',
  SUCCESS: 'SUCCESS',
  ERROR: 'ERROR',
};
