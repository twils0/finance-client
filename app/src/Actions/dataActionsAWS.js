import { actionTypes } from '../Constants/dataConstantsAWS';

export const setAWSStatus = (payload) => {
  if (!Object.prototype.hasOwnProperty.call(payload, 'status')) {
    throw new Error(`Please enter a value for the 'status' key - ${JSON.stringify(payload)}`);
  }

  return {
    type: actionTypes.SET_AWS_STATUS,
    payload,
  };
};

export const setAWSUser = (payload) => {
  if (!Object.prototype.hasOwnProperty.call(payload, 'user')) {
    throw new Error(`Please enter a value for the 'user' key - ${JSON.stringify(payload)}`);
  }
  if (typeof payload.user !== 'object' || payload.user.constructor === Array) {
    throw new Error(`Please provide an object value for the 'user' key - ${JSON.stringify(payload)}`);
  }

  return {
    type: actionTypes.SET_AWS_USER,
    payload,
  };
};

export const resetAWSState = () => ({ type: actionTypes.RESET_AWS_STATE });
