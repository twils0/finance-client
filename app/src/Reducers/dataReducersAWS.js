import { combineReducers } from 'redux';

import { requestStatusTypes } from '../Constants/universalConstants';
import { actionTypes } from '../Constants/dataConstantsAWS';

export const initialState = {
  status: requestStatusTypes.IDLE,
  user: null,
};

export const status = (state = initialState.status, action) => {
  switch (action.type) {
    case actionTypes.SET_AWS_STATUS:
      return action.payload.status;
    case actionTypes.RESET_AWS_STATE:
      return initialState.status;
    default:
      return state;
  }
};

export const user = (state = initialState.user, action) => {
  switch (action.type) {
    case actionTypes.SET_AWS_USER:
      return action.payload.user;
    case actionTypes.RESET_AWS_STATE:
      return initialState.user;
    default:
      return state;
  }
};

const dataReducersAWS = combineReducers({
  status,
  user,
});

export default dataReducersAWS;
