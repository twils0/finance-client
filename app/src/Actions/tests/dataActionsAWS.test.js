import configureMockStore from 'redux-mock-store';
import { initialState } from '../../Reducers/dataReducersAWS';
import { requestStatusTypes } from '../../Constants/universalConstants';
import { actionTypes } from '../../Constants/dataConstantsAWS';
import { setAWSStatus, setAWSUser, resetAWSState } from '../dataActionsAWS';

const middleware = [];
const mockStore = configureMockStore(middleware);

describe('dataActionsAWS', () => {
  describe('setAWSStatus', () => {
    it("fails when not provided a payload with a 'status' key", () => {
      const store = mockStore(initialState);

      const payload = {};

      expect(() => store.dispatch(setAWSStatus(payload))).toThrowErrorMatchingSnapshot();
    });

    it('creates the correct action with the correct payload', () => {
      const store = mockStore(initialState);

      const payload = { status: requestStatusTypes.SUCCESS };
      const expectedActions = [
        {
          type: actionTypes.SET_AWS_STATUS,
          payload,
        },
      ];

      store.dispatch(setAWSStatus(payload));

      const actions = store.getActions();

      expect(actions).toEqual(expectedActions);
    });
  });

  describe('setAWSUser', () => {
    it("fails when not provided a payload with an 'user' key", () => {
      const store = mockStore(initialState);

      const payload = {};

      expect(() => store.dispatch(setAWSUser(payload))).toThrowErrorMatchingSnapshot();
    });

    it("fails when provided a payload with an 'user' key that is not an object", () => {
      const store = mockStore(initialState);

      const payload = { user: [] };

      expect(() => store.dispatch(setAWSUser(payload))).toThrowErrorMatchingSnapshot();
    });

    it('creates the correct action with the correct payload', () => {
      const store = mockStore(initialState);

      const payload = { user: {} };
      const expectedActions = [
        {
          type: actionTypes.SET_AWS_USER,
          payload,
        },
      ];

      store.dispatch(setAWSUser(payload));

      const actions = store.getActions();

      expect(actions).toEqual(expectedActions);
    });
  });

  describe('resetAWSState', () => {
    it('creates the correct action', () => {
      const store = mockStore(initialState);

      const expectedActions = [
        {
          type: actionTypes.RESET_AWS_STATE,
        },
      ];

      store.dispatch(resetAWSState());

      const actions = store.getActions();

      expect(actions).toEqual(expectedActions);
    });
  });
});
