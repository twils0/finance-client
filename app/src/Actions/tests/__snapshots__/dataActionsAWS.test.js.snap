// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dataActionsAWS setAWSStatus fails when not provided a payload with a 'status' key 1`] = `"Please enter a value for the 'status' key - {}"`;

exports[`dataActionsAWS setAWSUser fails when not provided a payload with an 'user' key 1`] = `"Please enter a value for the 'user' key - {}"`;

exports[`dataActionsAWS setAWSUser fails when provided a payload with an 'user' key that is not an object 1`] = `"Please provide an object value for the 'user' key - {\\"user\\":[]}"`;
